### PR TITLE
Bug 2114504: flip menu vertically when it reaches the boundary

### PR DESCRIPTION
**Pull request by SanjalKatiyar**
**Before:**
Screenshot from 2022-09-16 19-38-06
**After:**
Screenshot from 2022-09-16 19-30-12
Screenshot from 2022-09-16 19-30-22

### CODE REVIEW

**Review by GowthamShanmugam — APPROVED** [review]
LGMT

**Comment by SanjalKatiyar** [comment]
> /lgtm cancel

this PR works totally fine and we can merge this as well if we want.
cancelled lgtm because PF also provides a prop `isFlipEnabled` for a Dropdown, but passing that prop is not working as expected. In sync with PF team as it's a known bug from PF side.

**Comment by GowthamShanmugam** [comment]
> isFlipEnabled

ack

**Comment by SanjalKatiyar** [comment]
> > /lgtm cancel
> 
> this PR works totally fine and we can merge this as well if we want. cancelled lgtm because new version of PF also provides a prop `isFlipEnabled` for a Dropdown, but passing that prop is not working as expected. In sync with PF team as it's a known bug from PF side.

@GowthamShanmugam @bipuladh we can go ahead and merge this PR for now (until it is fixed in newer version of PF).
PF has opened an issue on there end https://github.com/patternfly/patternfly-react/issues/8013 to fix the bug with `isFlipEnabled` prop !

File: packages/shared/kebab/kebab.tsx [hunk 1]
```diff
@@ -159,7 +178,10 @@ export const Kebab: React.FC<KebabProps> = ({
   }, [t, customKebabItemsMap, resourceLabel]);
 
   const toggle = React.useMemo(() => {
-    const onToggle = () => setOpen((open) => !open);
+    const onToggle = (_, e) => {
+      eventRef.current = e;
```
Comment (bipuladh) [inline packages/shared/kebab/kebab.tsx:182]: Could we handle all the logic in `useLayoutEffect` inside here itself?

Comment (SanjalKatiyar) [reply packages/shared/kebab/kebab.tsx:182]: that will not work, when this `onToggle` is called at that time event does not has a reference to the popper/dropdown items (cause they are not yet mounted to DOM)... only after the DOM is updated with the re-render which `onToggle` is causing, then only we can access the newly mounted element. Hence handling in a hook.

**Pull request merged**